osu.Game/Skinning/LegacyManiaSkinDecoder.cs

```diff
@@ -120,37 +120,38 @@ private void flushPendingLines()
                         break;
 
                     case "WidthForNoteHeightScale":
-                        currentConfig.WidthForNoteHeightScale = (float.Parse(pair.Value, CultureInfo.InvariantCulture)) * LegacyManiaSkinConfiguration.POSITION_SCALE_FACTOR;
+                        currentConfig.WidthForNoteHeightScale = Parsing.ParseFloat(pair.Value) * LegacyManiaSkinConfiguration.POSITION_SCALE_FACTOR;
                         break;
 
-                    case string when pair.Key.StartsWith("Colour", StringComparison.Ordinal):
+                    case { } when pair.Key.StartsWith("Colour", StringComparison.Ordinal):
```

Comment:
Note to self/others: while we don't use this brace match syntax anywhere else (and I'd rather keep it that way), it can't be easily replaced as replacing with `ReadOnlySpan<char>` results in a "source expression always matches provided pattern" inspection.

Reply:
`case _` don't work here either. The `_` is resolved as a symbol instead of discard pattern.

Reply:
Honestly I'd rather just not use `switch-case-when` for these and split out into individual `is` / `if` / whatever.

Reply:
There's an even worse expression with list pattern:
```
case ['C', 'o', 'l', 'o', 'u', 'r', ..]:
```